### PR TITLE
Add pre-commit checks

### PR DESCRIPTION
**Pull request by egeakman**
- Add pre-commit checks
- Add lint action
- Fix pre-commit and sphinx-lint complaints

----
:books: Documentation preview :books:: https://docs-community--87.org.readthedocs.build/en/87/

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/sphinx-contrib/sphinx-lint
+    rev: v0.6.8
+    hooks:
+      - id: sphinx-lint
+
+  - repo: https://github.com/psf/black
```
Comment (hugovk) [inline .pre-commit-config.yaml:7]: Using this mirror lets us use mypyc-compiled black, which is about 2x faster:

https://black.readthedocs.io/en/stable/integrations/source_version_control.html

```suggestion
  - repo: https://github.com/psf/black-pre-commit-mirror
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/sphinx-contrib/sphinx-lint
```
Comment (hugovk) [inline .pre-commit-config.yaml:2]: Let's put sphinx-lint last, it's slower to run so I prefer to get faster feedback from the others first.

File: requirements.txt [hunk 3]
```diff
@@ -3,6 +3,7 @@ myst-parser
 furo
 sphinx-copybutton
 sphinx-autobuild
+pre-commit
```
Comment (hugovk) [inline requirements.txt:6]: We don't need to install pre-commit to build the docs.

```suggestion
```

Instead, let's add a `.github/workflows/lint.yml`:

```yml
name: Lint

on: [push, pull_request, workflow_dispatch]

permissions:
  contents: read

jobs:
  lint:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v4
        with:
          python-version: "3.x"
      - uses: pre-commit/action@v3.0.0
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  - repo: https://github.com/psf/black
+    rev: 23.9.1
+    hooks:
+      - id: black
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```
Comment (hugovk) [inline .pre-commit-config.yaml:16]: Some other handy ones:
```suggestion
    hooks:
      - id: check-case-conflict
      - id: check-merge-conflict
      - id: check-yaml
      - id: end-of-file-fixer
      - id: trailing-whitespace
```

**Review by hugovk — COMMENTED** [review]
Thanks, a few suggestions :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (egeakman) [reply .pre-commit-config.yaml:7]: Thanks! Didn't know this existed.

**Comment by egeakman** [comment]
This PR got a little big in terms of file count. Should I open a seperate PR for the fixes after this one?

cc: @hugovk

File: docs/monthly-meeting/2023-02.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ Please take a second to read through it!
 ## Reports and celebrations
 
 > 60 second updates on things you have been up to, questions you have, or developments you think people should know about. Please add yourself, and if you do not have an update to share, you can pass.
-> 
+>
```
Comment (hugovk) [inline docs/monthly-meeting/2023-02.md:30]:
```suggestion
```

File: .github/workflows/lint.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
```
Comment (hugovk) [inline .github/workflows/lint.yaml:16]: `3.x` is "the latest stable version of Python", currently 3.11, and we won't need to update it when 3.12 is out.

https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-a-python-version

```suggestion
          python-version: "3.x"
```

**Review by hugovk — COMMENTED** [review]
> This PR got a little big in terms of file count. Should I open a seperate PR for the fixes after this one?

No, this is fine, there's not too many and they're small changes.

**Review by hugovk — APPROVED** [review]
Thank you!

**Pull request merged**